pkgs/development/tools/build-managers/gradle/update-deps.nix

```diff
@@ -0,0 +1,143 @@
+{ lib
+, pkgs
+, runtimeShell
+, stdenv
+, srcOnly
+, writeTextFile
+, writeShellScript
+, path
+, bubblewrap
+, coreutils
+, curl
+, jq
+, mitm-cache
+, nix
+, openssl
+, procps
+, python3
+}:
+
+let
+  getPkg = attrPath:
+    lib.getAttrFromPath
+      (lib.splitString "." (toString attrPath))
+      pkgs;
+in
+lib.makeOverridable (
+# the derivation to update
+{ pkg ? getPkg attrPath
+, pname ? null
+, attrPath ? pname
+# bwrap flags (this will be put in bash as-is)
+# this is relevant for downstream users
+, bwrapFlags ? "--ro-bind \"$PWD\" \"$PWD\""
+# deps path (relative to the package directory, or absolute)
+, depsPath ? "deps.json"
+# redirect stdout to stderr to allow the script to be used with update script combinators
+, silent ? true
+, useBwrap ? stdenv.isLinux
+}:
+
+let
+  keep = [ "MITM_CACHE_HOST" "MITM_CACHE_PORT" "MITM_CACHE_ADDRESS" "MITM_CACHE_CA" "MITM_CACHE_CERT_DIR" ];
+  gradleScript = writeShellScript "gradle-commands.sh" ''
+    set -eo pipefail
+    export http_proxy="$MITM_CACHE_ADDRESS"
+    export https_proxy="$MITM_CACHE_ADDRESS"
+    export SSL_CERT_FILE="$MITM_CACHE_CA"
+    export NIX_SSL_CERT_FILE="$MITM_CACHE_CA"
+    export GRADLE_USER_HOME="$(${coreutils}/bin/mktemp -d)"
+    export IN_GRADLE_UPDATE_DEPS=1
+    trap "${coreutils}/bin/rm -rf '$GRADLE_USER_HOME'" SIGINT SIGTERM ERR EXIT
```

Comment:
Didn't you convert this to `$PATH` at some point?

Reply:
this is a separate script that doesn't use PATH, I use the full coreutils path here because `stdenv` code could affect `PATH` in uncontrollable ways

Reply:
What I meant is `export PATH=${coreutils}/bin:"$PATH"`. That works regardless of what other path shenanigans are done.

Reply:
it doesn't, because the `trap` is executed after the derivation code, not before, so its PATH will be evaluated after executing the derivation code